internal/dao/pgx/stat_pgx.go

```diff
@@ -20,31 +20,31 @@ func getStatDao(ctx context.Context) dao.StatDao {
 	return &statDao{}
 }
 
-func (x *statDao) getUsage(ctx context.Context, interval string) ([]*models.UsageStat, error) {
+func (x *statDao) getUsage(ctx context.Context, iStart, iEnd string) ([]*models.UsageStat, error) {
 	query := `select provider, 'success' as result, count(provider) as count
 	from reservations
-	where created_at >= now() - cast($1 as interval)
+	where created_at between now() - cast($1 as interval) and now() - cast($2 as interval)
 	  and success = true
 	group by provider
 
 	union all
 
 	select provider, 'failure' as result, count(provider) as count
 	from reservations
-	where created_at >= now() - cast($1 as interval)
+	where created_at between now() - cast($1 as interval) and now() - cast($2 as interval)
 	  and success = false
 	group by provider
 
 	union all
 
 	select provider, 'pending' as result, count(provider) as count
 	from reservations
-	where created_at >= now() - cast($1 as interval)
+	where created_at between now() - cast($1 as interval) and now() - cast($2 as interval)
 	  and success is null
 	group by provider`
 
 	var result []*models.UsageStat
-	rows, err := db.Pool.Query(ctx, query, interval)
+	rows, err := db.Pool.Query(ctx, query, iStart)
```

Comment:
```suggestion
	rows, err := db.Pool.Query(ctx, query, iStart, iEnd)
```
I think you want to do this. Without the second argument, it fails.

Reply:
Good catch, see, never push untested code :-)

Fixed